modules/core/react/lib/InputProvider.tsx

```diff
@@ -170,9 +175,17 @@ export default class InputProvider extends React.Component<InputProviderProps, I
   provideIntent = this.props.provideIntent;
 
   componentDidMount() {
+    const {container} = this.props;
+    if (container) {
+      // bail early if input provider is already handling this
+      if (container.classList.contains(containerClass)) {
```

Comment:
We can just change `this.ref.current.parentElement.closest(`.${containerClass}`)` to `container.closest('[data-whatinput]')` for the same effect (`closest` checks the element itself and all ancestors).